ex4/ex4.c

```diff
@@ -10,6 +10,18 @@
 int main(int argc, char* argv[])
 {
     // Your code here    
+    int rc = fork();
 
+    if (rc < 0) { // Fail -> exit
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+      } else if (rc == 0) { // Child
+        printf("Child\n");
+        execl("ls", NULL);
+      } else { // Parent
+        int wc = waitpid(rc, NULL, 0);
```

Comment:
`waitpid` here is an implicit declaration (meaning the compiler has no way of telling whether you're using the function correctly), I posted some information regarding that for ex3 as well.